### PR TITLE
feat req: increase composer's height to 48px

### PR DESCRIPTION
**Pull request by daraksha-dk**
### Details

https://github.com/Expensify/App/issues/15819#issuecomment-1514530783
Increased the composer's height to 48px and the Send button's size to 40px.

### Fixed Issues

$ https://github.com/Expensify/App/issues/15819
PROPOSAL: https://github.com/Expensify/App/issues/15819#issuecomment-1514530783

### Tests

1. Open the app
2. Go to any chat
3. Verify that the height of composer is 48px
4. Verify that the size of Send and Emoji button is 40x40
5. Verify that Actions icon's size is 32x40 ("+") 

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open the app
2. Go to any chat
3. Verify that the height of composer is 48px
4. Verify that the size of Send and Emoji button is 40x40
5. Verify that Actions icon's size is 32x40 ("+") 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/71884990/2a2df17c-505a-4c95-bcee-bc34908b3f32

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/71884990/5bdb3d31-47c5-4037-a8fa-d754c56b4e34

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/71884990/99ac0613-9f7d-4b23-844b-c8c0bddacb42

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/71884990/90e13e0d-322d-4478-afa0-0d4ef4449e8a

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/71884990/04ddc455-b74f-4c9b-83d3-0a0dadc10cfc

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/71884990/199d6d57-7b4e-4e2e-9d81-3962fc5ff43f

</details>

### CODE REVIEW

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/ddc05481-9cc8-4fd2-84d1-865e033c0406
</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/25876548/2ae94813-c500-4e3f-845f-982db73ddfe3



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/25876548/fde20804-104d-487a-a42e-b186ff839941


</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/c314a894-0850-402f-894d-cf0e5cb302dc

</details>

<details>
<summary>iOS</summary>



https://github.com/Expensify/App/assets/25876548/6d6c0b5a-12fa-432b-a88b-f3728cedc15d


</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/2307b68f-ee45-433d-a567-a0abbedc19cc


</details>

**Review by sobitneupane — COMMENTED** [review]
@daraksha-dk 
Meanwhile there are still some issues:

Screenshot 2023-06-21 at 11 36 44

Inconsistent change in height on adding new line.

https://github.com/Expensify/App/assets/25876548/f0aa715a-6c3e-4904-8390-b330895cf1d5

**Comment by shawnborton** [comment]
Nice, this looks pretty good to me.

**Comment by shawnborton** [comment]
Can you spin up a test build when you are ready so I can try it on my end too?

**Comment by shawnborton** [comment]
Okay cool, let's try it!

**Comment by MariaHCD** [comment]
Tried to get those builds but no luck for me either :(

**Review by sobitneupane — COMMENTED** [review]
Composer height increases only by 4px on adding first new line. Padding decreases.

https://github.com/Expensify/App/assets/25876548/f6080017-b41d-4760-89d1-c4a4a4de732a

Similar was the base before as well. But the decrease in padding was quite small and looks as if it is expected.

**Before the change:**

https://github.com/Expensify/App/assets/25876548/9ce2b3f0-eeed-4852-9a85-3f8580f919c1

What should we do for this @shawnborton?

I also noticed that on adding second line, the padding again changes (may to adjust Expand button) and changes again on adding third line. It is an existing issue.

**Comment by mountiny** [comment]
Build is running here https://github.com/Expensify/App/actions/runs/5343785571

**Comment by shawnborton** [comment]
Tested quickly and I think this feels pretty good to me.

**Review by MariaHCD — DISMISSED** [review]
Code LGTM! Over to you, @sobitneupane

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/100f5989-bc2a-4c01-a32b-af1d90ba2c60

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/25876548/15ef86ca-cfca-491e-b4eb-20ac40ce2dbc

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/d9d54ac7-07bb-4b81-a814-f9844d577314

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/0f99082c-16a1-447a-9c4a-442d25d44c16

</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/25876548/495d3df4-a3b2-4ee1-8cf6-b00229134809

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/06926e72-6653-486a-b425-8ef151e5918a

</details>

File: src/pages/home/report/ReportActionCompose.js [hunk 1]
```diff
@@ -953,7 +953,7 @@ class ReportActionCompose extends React.Component {
                                                     style={[
                                                         styles.dFlex,
                                                         styles.flexColumn,
-                                                        isFullComposerAvailable || this.props.isComposerFullSize ? styles.justifyContentBetween : styles.justifyContentEnd,
+                                                        isFullComposerAvailable || this.props.isComposerFullSize ? styles.justifyContentBetween : styles.justifyContentCenter,
```
Comment (sobitneupane) [inline src/pages/home/report/ReportActionCompose.js:956]: `Add action` (`+`) icon is moving up and down due to this change. Let's revert it back to `justifyContentEnd`. And change height of style.composerSizeButton to 40 if it makes sense and solves the issue you're trying to solve by this change.

**Comment by mountiny** [comment]
The desktop build failure is unrealted.

File: src/pages/home/report/ReportActionCompose.js (hunk 1 above)
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: It looks bad than before. There is a huge change in height on adding 3rd line but no change at all on 4th.
https://github.com/Expensify/App/assets/25876548/9f56367a-5515-45c9-8bc9-bf9c8f3edcaa

**Comment by sobitneupane** [comment]
@shawnborton Are we okay with following changes?

Following small changes are resulted by this PR.

1. Add action '+' moves slightly up on adding first new line and moves slightly down on adding second new line. 

https://github.com/Expensify/App/assets/25876548/43cdf22d-7ba2-475d-bd4b-37b63b0301db

We tired to solve this issue by changing height of Add action '+' but it resulted https://github.com/Expensify/App/pull/21131#discussion_r1240080861 issue.

2. On adding new line Height of the composer changes in following pattern:

    With change in this PR:
    48px -> 52 px -> 78 px -> 92px -> +20 on addition of each line afterward
    
    Previously:
    40px -> 52px -> 78px -> 92px -> +20 on addition of each line afterward.

**Comment by shawnborton** [comment]
> Add action '+' moves slightly up on adding first new line and moves slightly down on adding second new line.

I think this is fine and is expected

**Comment by shawnborton** [comment]
> (4) Apply paddingBottom of 10px for + icon - This will help the behaviour stay as similar to before as possible.
Con: The symmetry of the design is disrupted on the third line and beyond, due to the bottom being 4 pixels longer. To rectify this, an option could be to add 4 pixels to the top as well to achieve symmetry but that will further increase the height of composer.

I can't quite understand this. The video you shared for this seems to look fine to me?

File: src/pages/home/report/ReportActionCompose.js (hunk 1 above)
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: @daraksha-dk Thanks for testing different scenarios, let's revert back to `justifyContentCenter `

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by MariaHCD — APPROVED** [review]
LGTM, leaving the final review to @shawnborton

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
Looks good to me too, but where can I find the most up-to-date screenshots? Especially ones that show the textarea expanding to multi lines. If we have new screenshots, let's add them to the original comment please!

**Comment by shawnborton** [comment]
Sounds good, I'll merge this one then.

**Pull request merged**